Fix Blazor Maui projects on VS Mac

davidwengier opened the pull request:
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1544050 and maybe even https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1315748/ if I get lucky?

davidwengier commented on src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs:
```diff
@@ -204,24 +203,17 @@ protected void RemoveDocument(HostProject hostProject, string filePath)
         }
 
-        var baseIntermediateOutputPathValue = projectProperties.GetValue(BaseIntermediateOutputPathPropertyName);
+        var baseIntermediateOutputPathValue = DotNetProject.BaseIntermediateOutputPath.ToString();
```
This fixes the issue, because VS Mac will give us the fully qualified path from this property, so everything works

ryanbrandenburg replied:
So `projectProperties.GetValue(BaseIntermediateOutputPathValue)` is a relative path then?

davidwengier commented on src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/MacRazorProjectHostBase.cs:
```diff
@@ -264,15 +256,12 @@ protected void RemoveDocument(HostProject hostProject, string filePath)
-        }
+        var projectDirectory = projectProperties.HasProperty(MSBuildProjectDirectoryPropertyName)
+            ? projectProperties.GetValue(MSBuildProjectDirectoryPropertyName)
+            : DotNetProject.BaseDirectory.ToString();
```
This is belt and braces.. the current code is failing because `MSBuildProjectDirectory` is null for MAUI projects on Mac, so falling back to the `BaseDirectory` makes sense, but also the above change means it won't get here.

davidwengier commented:
I think there might be more to this.. this doesn't actually fix the problem, but would fix where we write the `project.razor.vs.json` file to, except it seems we don't write that file when not in LSP mode, so something else is going on here, presumably with component discovery itself. Still debugging...

davidwengier commented:
Updated the PR with what should be the actual fix. In https://github.com/dotnet/project-system/pull/7417 I allowed any item type to be a Razor file, because various projects (OrchardCore, Bing, Maui projects) used different item types to express things. This change never made it to VS Mac. It now has.

While there, I also removed the case sensitivity, to match the changes made to the compiler in https://github.com/dotnet/razor/pull/7934

Doing a CI build to validate on Mac now.

davidwengier commented:
> Doing a CI build to validate on Mac now.

NOT IF I DON'T REMEMBER TO PUSH TO THE RIGHT REMOTE, I'M NOT!!! 
😡😡😡😡😡

davidwengier commented on src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/DefaultMacRazorProjectHost.cs:
```diff
@@ -115,18 +114,12 @@ internal static bool IsRazorDocumentItem(IMSBuildItemEvaluated item)
         }
 
-        if (!item.Include.EndsWith(".razor", StringComparison.Ordinal) && !item.Include.EndsWith(".cshtml", StringComparison.Ordinal))
+        if (!item.Include.EndsWith(".razor", StringComparison.OrdinalIgnoreCase) && !item.Include.EndsWith(".cshtml", StringComparison.OrdinalIgnoreCase))
```
The compiler recently made the same change, as noted in another comment, so figured we should follow suit. In general macOS is case sensitive though, so a `.Razor` file is possible, and the compiler change was from a customer report

davidwengier commented on src/Razor/src/Microsoft.VisualStudio.Mac.LanguageServices.Razor/ProjectSystem/DefaultMacRazorProjectHost.cs:
```diff
@@ -115,18 +114,12 @@ internal static bool IsRazorDocumentItem(IMSBuildItemEvaluated item)
             throw new ArgumentNullException(nameof(item));
         }
 
-        if (item.Name != ContentItemType)
```
It behaves the same, but we made a change to not rely on a specific item type at all. In Maui projects .razor files are not Content items
